nca/Utils/ExplTracker.py

```diff
@@ -0,0 +1,426 @@
+            When there is no connection, we list all the policies that affect the peers, so the user may have all the info
+            to find problems.
+
+            _is_active - flag for checking if expl' was activated
+
+            all_conns - all the calculated connection. This is used to shortcut the check if 2 peers are connected or not
+
+            all_peers - all the peers. This is used for the explain-all feature.
+
+            ep - endpoints configurations (use either full_name for pod mode, or workload_name for deployments mode)
+    """
+
+    def __init__(self, ep=''):
+        self.ExplDescriptorContainer = {}
+        self.ExplPeerToPolicyContainer = {}
+        self._is_active = False
+        self.all_conns = None
+        self.all_peers = None
+        self.ep = ep
+
+        self.add_item('', 0, 'Default-Policy')
+
+    class ExplPolicies:
+        """
+        ExplPolicies holds the policies affecting peers in relation to other peers.
+        That is, for each peer it holds all the peers in its egress and ingress and the policies
+        that has effect on the connection to that peers.
+        """
+
+        def __init__(self):
+            self.egress_dst = {}
+            self.ingress_src = {}
+            self.all_policies = set()
+
+        @staticmethod
+        def _add_policy(peer_set, peer_list, policy_name):
+            """
+            Adds a policy to the list of affecting policies, for each peer in the peer_set
+            :param PeerSet peer_set: a set of peers to add the policy to
+            :param dict peer_list: a list of peers that holds the policies affecting them
+            :param str policy_name: the policy to add
+            """
+            for peer in peer_set:
+                peer_name = peer.full_name()
+                if not peer_list.get(peer_name):
+                    peer_list[peer_name] = set()
+                # We don't want Default-Policy if we have any other policy,
+                # so we first remove it and then add the policy (even if we currently add
+                # the Default-Policy itself).
+                peer_list[peer_name].discard('Default-Policy')
+                peer_list[peer_name].add(policy_name)
+
+        def add_policy(self, policy_name, egress_dst, ingress_src):
+            """
+            Adds a given policy to the relevant peer lists (egress list, ingress list)
+            :param str policy_name: name of the policy
+            :param PeerSet egress_dst: the set of egress destinations peers to add the policy too
+            :param PeerSet ingress_src: the set of ingress source peers to add the policy too
+            """
+            self.all_policies.add(policy_name)
+
+            if egress_dst:
+                self._add_policy(egress_dst, self.egress_dst, policy_name)
+
+            if ingress_src:
+                self._add_policy(ingress_src, self.ingress_src, policy_name)
```

Comment:
- confused by method names `_add_policy` vs `add_policy`
- confusing param names (e.g. `peer_list` would expect to be a `list` but is a `dict` + its documentation does not explain it's a map from ...  to .... )

Reply:
Done (changed the _add_policy name so it will be clearer. Also documented it better).